From da6eac0c2df1d556c53831512cb0b22ade2b3772 Mon Sep 17 00:00:00 2001
From: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>
Date: Wed, 8 Feb 2017 00:13:45 +0000
Subject: [PATCH] configure: Add --enable-autogopath flag

This patch adds --eneble-autogopath flag to configure.
This allow to build cc-oci-runtime without set the GOPATH
manually.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>
---
 Makefile.am  | 18 ++++++++++++++++--
 configure.ac | 11 +++++++++++
 2 files changed, 27 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4b98caa..7d057fe 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,6 +25,7 @@ AUTOMAKE_OPTIONS = parallel-tests
 
 CHECK_DEPS =
 FUNCTIONAL_TESTS_DEPS =
+PROXY_DEPS =
 
 # Has a value if building in a git tree
 GIT_COMMIT := $(shell git rev-parse HEAD 2>/dev/null)
@@ -250,8 +251,21 @@ systemdservicedir   = $(systemdsystemunitdir)
 systemdservice_DATA = $(systemdservice_files)
 endif
 
+if AUTOGOPATH
+export GOPATH := $(CURDIR)/.gopath
+PKG_LINK := $(CURDIR)/.gopath/src/github.com/01org/cc-oci-runtime
+
+$(PKG_LINK):
+		rm -rf $(CURDIR)/.gopath
+		mkdir -p  $(CURDIR)/.gopath/src/github.com/01org
+		ln -sfn $(CURDIR) $(PKG_LINK)
+		cp -r $(CURDIR)/vendor/* $(CURDIR)/.gopath/src
+
+PROXY_DEPS += $(PKG_LINK)
+endif
+
 proxy_ldflags = "-X main.DefaultSocketPath=$(localstatedir)/run/cc-oci-runtime/proxy.sock"
-cc-proxy: $(cc_proxy_sources)
+cc-proxy: $(cc_proxy_sources)  $(PROXY_DEPS)
 	$(AM_V_GO)go build -o $@ -ldflags=$(proxy_ldflags) $(srcdir)/proxy
 
 cc_proxy_sources =			\
@@ -285,7 +299,7 @@ func main() {
 endef
 export CONTENT
 
-check-proxy:
+check-proxy: $(cc_proxy_sources) | $(PROXY_DEPS)
 	$(AM_V_GEN)go_test_extra_args="" ; \
 	echo "$$CONTENT">./racetest.go ; \
 	if go build -race ./racetest.go; then \
diff --git a/configure.ac b/configure.ac
index 186ecc4..5011444 100644
--- a/configure.ac
+++ b/configure.ac
@@ -302,6 +302,17 @@ AC_ARG_WITH([cc-image],
     ]
 )
 
+AC_ARG_ENABLE([autogopath],
+	AS_HELP_STRING([--enable-autogopath], [enable autogopath @<:@default=false@:>@]),
+	[case "${enableval}" in
+		yes) autogopath=true ;;
+		 no) autogopath=false ;;
+		  *) AC_MSG_ERROR([bad value ${enableval} for --enable-autogopath]) ;;
+	esac],
+	[autogopath=false]
+)
+AM_CONDITIONAL([AUTOGOPATH], [test x$autogopath = xtrue])
+
 AC_OUTPUT
 
 dnl === Summary ===============================================================
